### DISCUSSION TITLE
Does PnP Core (SDK) have methods to interact with the Site Design (REST) API?

### DISCUSSION THREAD

**Discussion by stevedDev** [pnp/pnpcore#1495] [General] [answered]
Hi guys,

I need to implement some Site design manipulation code in one of our projects that already uses PnP Core, but I haven't been able to find anything within the current documentation/sources that seems to use that API at all (apart from PnPJS).
So basically asking whether it is already covered or should I just hit the REST APIs directly, or maybe look to add it to PnP Core?

Regards
Steve

**Comment by stevedDev** [comment]
I have found that I can call the REST API using the ApiRequest mechanism;

var apiRequest = new ApiRequest( HttpMethod.Post, ApiRequestType.SPORest, "_api/Microsoft.SharePoint.Utilities.WebTemplateExtensions.SiteScriptUtility.ApplySiteDesign", rqstBody );

And although that call "seems" to work (returns a 200 status), it doesn't apply the site design, which makes me wonder whether the Site Design API doesn't work under app-only?

**Reply by jansenbe** [reply, marked as answer]
@stevedDev : we do not yet have support for working with site designs in Core SDK, using the custom API route (https://pnp.github.io/pnpcore/using-the-sdk/basics-customapirequests.html) is a good path forward. We also welcome PRs bringing site design support :-)

**Reply by stevedDev** [reply]
Hi Bert,

Thank you I managed to resolve the issue  by calling the "AddSiteDesignTaskToCurrentWeb" api endpoint rather than "ApplySiteDesign", not entirely sure why there is a difference, but as usual no time to investigate further.
And I will at sometime in the future get around to look to raise PRs to enhance things rather than just raising issues, just never seem to have enough "free" time ;)

Thanks again
